checks/cs50/2017/x/finance/checks.py

```diff
@@ -1,13 +1,305 @@
 import os
 import re
+import requests
 import sys
 
 sys.path.append(os.getcwd())
-from check50 import TestCase, Error, check
+from check50 import File, TestCase, Error, check
+from selenium import webdriver
 
 class Finance(TestCase):
+
+    BASE_PATH = "http://127.0.0.1:5000"
+    
+    def server_up(self):
+        env = {
+            "FLASK_APP": "application.py"
+        }
+
+        self.server = self.spawn("flask run", env=env)
+        self.driver = webdriver.PhantomJS()
+
+    def server_down(self):
+        self.driver.close()
+        self.server.kill()
+
+    def override_lookup(self):
+        """Replaces student's lookup function with a deterministic one."""
+        self.append_code("helpers.py", File("lookup.py"))
+    
+    def register(self, username, password, confirmation, outcome):
+        """Checks that registering results in success or failure."""
+
+        # go to register page
+        self.driver.get("{}/register".format(Finance.BASE_PATH))
+        fields = self.driver.find_elements_by_tag_name("input")
+    
+        # check for input fields
+        if len(fields) == 0:
+            raise Error("No available input fields to register.")
+        
+        # fill in name, password, confiramtion
+        fields_filled = 0
+        for field in fields:
+            name = field.get_attribute("name")
+            if name == "username":
+                fields_filled += 1
+                field.send_keys(username)
+            elif name == "password":
+                fields_filled += 1
+                field.send_keys(password)
+            elif name == "confirmation":
+                fields_filled += 1
+                field.send_keys(confirmation)
+        if fields_filled != 3:
+            raise Error("Could not fill in username and/or password.")
+    
+        # search for register button
+        buttons = self.driver.find_elements_by_tag_name("button")
+        register_button = None
+        for button in buttons:
+            if button.text == "Register":
+                register_button = button
+        if register_button == None:
+            raise Error("Could not find register button.")
+        register_button.click()
+
+        if "Apology" in self.driver.title and outcome:
+            raise Error("Expected registration success, not rejection.")
+        elif "Apology" not in self.driver.title and not outcome:
```

Comment:
Feels like we should also be abstracting away `driver`, maybe calling it `document` or such?